src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/FullScreen/PlatformFullScreenEffect.android.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.ComponentModel;
+using Android.OS;
+using Android.Views;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android;
+using static Xamarin.CommunityToolkit.Effects.FullScreenEffect;
+using static Xamarin.CommunityToolkit.PlatformConfiguration.AndroidSpecific.FullScreenEffect;
+using Effects = Xamarin.CommunityToolkit.Android.Effects;
+
+[assembly: ExportEffect(typeof(Effects.FullScreenEffectRouter), nameof(FullScreenEffectRouter))]
+
+namespace Xamarin.CommunityToolkit.Android.Effects
+{
+	public class FullScreenEffectRouter : PlatformEffect
+	{
+		bool isDetaching;
+
+		protected override void OnAttached()
+		{
+			// should we use WeakEventManager instead?
+			(Element as Page)!.Disappearing += FullScreenEffectRouter_Disappearing;
```

Comment:
Instead of casting Element as Page everywhere, let's just do:

```csharp
               Page PageElement => (Page)Element;

		protected override void OnAttached()
		{
			PageElement.Disappearing += FullScreenEffectRouter_Disappearing;
                        // ....
                 }
```